.github/workflows/main.yml

```diff
@@ -3,18 +3,169 @@ on:
     branches:
       - main
   pull_request:
-  
+
+env:
+  MIX_ENV: test
+
 jobs:
+  lint:
```

Comment:
CI is just doing a lot more checks now, like checking Dialyzer, unused dependencies, formatting, and so on.